config.go

```diff
@@ -12,168 +12,200 @@
 // limitations under the License.
 
+//go:generate paramgen -output config_paramgen.go Config
+
 package generator
 
 import (
 	"errors"
 	"fmt"
-	"math"
-	"strconv"
 	"strings"
 	"time"
-)
 
-const (
-	RecordCount   = "recordCount"
-	ReadTime      = "readTime"
-	SleepTime     = "burst.sleepTime"
-	GenerateTime  = "burst.generateTime"
-	FormatType    = "format.type"
-	FormatOptions = "format.options"
-	Operation     = "operation"
-
-	FormatRaw        = "raw"
-	FormatStructured = "structured"
-	FormatFile       = "file"
+	"github.com/conduitio/conduit-connector-generator/internal"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"golang.org/x/time/rate"
 )
 
-var (
-	knownFieldTypes = []string{"int", "string", "time", "bool"}
-	requiredFields  = []string{FormatType, FormatOptions}
+const (
+	FormatTypeRaw        = "raw"
+	FormatTypeStructured = "structured"
+	FormatTypeFile       = "file"
 )
 
-type RecordConfig struct {
-	FormatType    string
-	FormatOptions interface{}
-	Operation     string
+type Config struct {
+	Burst ConfigBurst `json:"burst"`
+	// Number of records to be generated (0 means infinite).
+	RecordCount int `json:"recordCount" validate:"gt=-1"`
+	// The time it takes to 'read' a record.
+	// Deprecated: use `rate` instead.
+	ReadTime time.Duration `json:"readTime"`
+	// The maximum rate in records per second, at which records are generated (0
+	// means no rate limit).
+	Rate float64 `json:"rate"`
+
+	// Configuration for default collection (i.e. records without a collection).
+	// Kept for backwards compatibility.
+	ConfigCollection
+	Collections map[string]ConfigCollection `json:"collections"`
 }
 
-func ParseRecordConfig(formatType, formatOptions, operation string) (RecordConfig, error) {
-	c := RecordConfig{
-		FormatOptions: make(map[string]interface{}),
-		Operation:     operation,
-	}
-	// check if it's a recognized format
-	switch formatType {
-	case FormatFile:
-		c.FormatType = FormatFile
-		if formatOptions == "" {
-			return RecordConfig{}, errors.New("file path not specified")
-		}
-		c.FormatOptions = formatOptions
-	case FormatStructured, FormatRaw:
-		fields, err := parseFields(formatOptions)
-		if err != nil {
-			return RecordConfig{}, fmt.Errorf("failed parsing fields: %w", err)
-		}
-		c.FormatType = formatType
-		c.FormatOptions = fields
-	default:
-		return RecordConfig{}, fmt.Errorf("unknown payload format %q", formatType)
-	}
+type ConfigBurst struct {
+	// The time the generator "sleeps" between bursts.
+	SleepTime time.Duration `json:"sleepTime"`
+	// The amount of time the generator is generating records in a burst. Has an
+	// effect only if `burst.sleepTime` is set.
+	GenerateTime time.Duration `json:"generateTime" default:"1s"`
+}
 
-	return c, nil
+type ConfigCollection struct {
+	// Comma separated list of record operations to generate. Allowed values are
+	// "create", "update", "delete", "snapshot".
+	Operation []string     `json:"operation" default:"create" validate:"required"`
```

Comment:
@reviewer - what do you think of renaming this to `operations`, since it accepts multiple?

Reply:
@lovromazgon Hmm... I can't think of a possibly reasonable argument, but my theory is that the most common use-case would be generating _just_ one operation, and making it plural could cause confusion around the expectation. However, if we want to make this consistent with how we've been approaching the other connector fields that accept multiple collections, it would make sense to rename this to `Operations`.

Reply:
Yeah, that's my reasoning as well, consistency. For instance, the kafka source now accepts `topics`, even though the average user probably only consumes a single topic. The plural parameter name also kinda serves as documentation, it tells you that you can specify multiple. That's why I'm leaning towards `operations` here.

Reply:
@lovromazgon agreed. Let's use `Operations` then!